go/vt/vtctl/workflow/server.go

```diff
@@ -3310,19 +3344,30 @@ func (s *Server) switchWrites(ctx context.Context, req *vtctldatapb.WorkflowSwit
 		}
 	}
 
-	// Need to lock both source and target keyspaces.
-	tctx, sourceUnlock, lockErr := sw.lockKeyspace(ctx, ts.SourceKeyspaceName(), "SwitchWrites")
+	// Lock the workflow and its source and target keyspaces.
+	lockName := fmt.Sprintf("%s/%s", ts.TargetKeyspaceName(), ts.WorkflowName())
+	ctx, workflowUnlock, lockErr := s.ts.LockName(ctx, lockName, "SwitchWrites")
```

Comment:
Should we not be applying the same TTL to this lock as we do to the keyspace lock?

Reply:
Named locks currently have no practical TTL, with hardcoded TTL of 24 hours just as a final safety net if somehow a lock is left in place by a still running session.